File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int min_bph = 1, max_bph = *max_element(piles.begin(), piles.end());
        int answer = 0;
        while (min_bph <= max_bph) {
            int mid = (max_bph + min_bph) / 2;
            long timeTaken = calculateTimeTaken(piles, mid);
            if (timeTaken <= h) {
                answer = mid;
                max_bph = mid - 1;
            } else {
                min_bph = mid + 1;
            }
        }
        return answer;
    }

    long calculateTimeTaken(vector<int>& piles, int bph) {
        long time = 0;
        for (int pile : piles) {
            time += ceil((double)pile / (double)bph);
        }
        return time;
    }
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.ts

function minEatingSpeed(piles: number[], h: number): number {
    let minBph = 1;
    let maxBph = Math.max(...piles);
    let answer = 0;
    while (minBph <= maxBph) {
        const mid = Math.floor(minBph + (maxBph - minBph) / 2);
        const timeTaken = calculateTimeTaken(piles, mid);
        if (timeTaken <= h) {
            answer = mid;
            maxBph = mid - 1;
        } else {
            minBph = mid + 1;
        }
    }
    return answer;
}

function calculateTimeTaken(piles: number[], bph: number): number {
    let time = 0;
    for (const pile of piles) {
        time += Math.floor((pile + bph - 1) / bph);
    }
    return time;
}
